mithril-aggregator/src/http_server/routes/epoch_routes.rs

```diff
@@ -179,6 +180,95 @@ mod tests {
         assert_eq!(message.next_cardano_transactions_signing_config, None);
     }
 
+    #[tokio::test]
+    async fn get_epoch_settings_message_retrieves_protocol_parameters_from_epoch_service() {
+        let current_epoch_settings = AggregatorEpochSettings {
+            protocol_parameters: ProtocolParameters::new(101, 10, 0.5),
+            ..AggregatorEpochSettings::dummy()
+        };
+        let next_epoch_settings = AggregatorEpochSettings {
+            protocol_parameters: ProtocolParameters::new(102, 20, 0.5),
+            ..AggregatorEpochSettings::dummy()
+        };
+        let upcoming_epoch_settings = AggregatorEpochSettings {
+            protocol_parameters: ProtocolParameters::new(103, 30, 0.5),
+            ..AggregatorEpochSettings::dummy()
+        };
+
+        let epoch_service = FakeEpochService::with_data(
+            Epoch(1),
+            &current_epoch_settings,
+            &next_epoch_settings,
+            &upcoming_epoch_settings,
+            &fake_data::signers_with_stakes(5),
+            &fake_data::signers_with_stakes(3),
+        );
+
+        let message = get_epoch_settings_message(
+            Arc::new(RwLock::new(epoch_service)),
+            SignedEntityConfig::dummy(),
+        )
+        .await
+        .unwrap();
+
+        // TODO: What we expect  ? see #1963 Discrepancy of protocol parameters in epoch settings and pending certificates routes
+        // https://github.com/input-output-hk/mithril/issues/1963
```

Comment:
This can be fixed now that #1963 is closed

Reply:
Done ✅